subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -1122,22 +1144,38 @@ static int scan_delegator_rem_src(struct bt_conn *conn,
 	state = &internal_state->state;
 
 	/* If conn == NULL then it's a local operation and we do not need to ask the application */
-	if (conn != NULL && (state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ ||
-			     state->pa_sync_state == BT_BAP_PA_STATE_SYNCED)) {
-		/* Unlock mutex to avoid potential deadlock on app callback */
-		err = k_mutex_unlock(&internal_state->mutex);
-		__ASSERT(err == 0, "Failed to unlock mutex: %d", err);
+	if (conn != NULL) {
+
+		if (scan_delegator_cp_cbs != NULL &&
+			scan_delegator_cp_cbs->remove_source != NULL) {
+			err = scan_delegator_cp_cbs->remove_source(conn, src_id);
+			if (err != BT_ATT_ERR_SUCCESS) {
+				int mutex_err;
+
+				LOG_DBG("Remove Source rejected with reason 0x%02x", err);
+				mutex_err = k_mutex_unlock(&internal_state->mutex);
+				__ASSERT(mutex_err == 0, "Failed to unlock mutex: %d", mutex_err);
+				return BT_GATT_ERR(err);
+			}
+		}
 
-		/* Terminate PA sync */
-		err = pa_sync_term_request(conn, &internal_state->state);
-		if (err != 0) {
-			LOG_DBG("PA sync term from %p was rejected with reason %d", (void *)conn,
-				err);
+		if (state->pa_sync_state == BT_BAP_PA_STATE_INFO_REQ ||
+			state->pa_sync_state == BT_BAP_PA_STATE_SYNCED) {
+			/* Unlock mutex to avoid potential deadlock on app callback */
+			err = k_mutex_unlock(&internal_state->mutex);
+			__ASSERT(err == 0, "Failed to unlock mutex: %d", err);
 
-			return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
-		}
+			/* Terminate PA sync */
+			err = pa_sync_term_request(conn, &internal_state->state);
+			if (err != 0) {
+				LOG_DBG("PA sync term from %p was rejected with reason %d", (void *)conn,
+					err);
 
-		k_mutex_lock(&internal_state->mutex, K_FOREVER);
+				return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
+			}
+
+			k_mutex_lock(&internal_state->mutex, K_FOREVER);
```

Comment:
Missing error check for `k_mutex_lock`. I am aware that you just moved it, but we should do an `_ASSERT(mutex_err == 0, "Failed to lock mutex: %d", mutex_err);` check here as well